Document new features and update examples throughout docs

TheRealPear opened the pull request:
First pull request of the year to fix examples throughout the docs as well as documenting new PGM features. For convenience, I've put them in the form of check boxes with a link to the relevant issue or pull request from other PGM repositories.

- [X] Update example maps with XML changes. These maps were already using 1.4.2, but I wanted to bring over the changes I made in the linked pull request. (PGMDev/Maps#1)
- [X] Update pre-existing XML blocks to use 1.4.2 syntax (see _Structures_, _Classes_, _Items_, _Kits_, _Projectiles_, _Main Map Element_, _Filters_, _Portals_, _Regions_, _Spawns_, _Control Points_, _Capture the Wool_, _Other Objectives_, and _Scoring_)
- [X] Document actions in kill rewards module (PGMDev/PGM#1251)
- [X] Document variant `<if>`/`<unless>` conditional attributes (PGMDev/PGM#1250)
- [X] Document constant `delete` attribute (PGMDev/PGM#1256)
- [x] Update offset filter section (PGMDev/PGM#1262)
- [x] Add a section on inline filters in Filters page (#99 PGMDev/PGM#1106 PGMDev/PGM#1257)
- [x] Document `exclusive` attribute in dummy variables (PGMDev/PGM#1245)
- [x] Document `score`, `with-team`, and `lives` variable (PGMDev/PGM#1220)

Please comment if I have missed any PGM features that was recently added that is yet to be documented on pgm.dev

CoWinkKeyDinkInc commented on docs/examples/airship-battle.mdx:
```diff
@@ -133,57 +133,36 @@ These regions reference the filters defined above and states where they will wor
 <regions>
     <cuboid id="red-spawn-protect" min="-42,87,3" max="-35,90,-36"/>
     <cuboid id="blue-spawn-protect" min="12,87,3" max="19,90,-36"/>
+    <complement id="deny-obsidian-area">
```
```suggestion
    <!-- Subtracts successive regions (blue-core and red-core) from the first region (map) -->
    <complement id="deny-obsidian-area">
```

TheRealPear commented:
Documented offset filter changes, `exclusive` attribute, and new variables.

TheRealPear commented:
Everything has been checked off and reviewed with Pablo

TheRealPear commented:
* Clarified more on module id and variable id limitations
* Added `{0}` example to countdown, after, and pulse filter section
* Changed variable examples to use underscore instead of dash for its ids
